### PR TITLE
Support for ghc-8.10.3

### PR DESCRIPTION
**Pull request by jneira**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -137,15 +137,20 @@ jobs:
...
+  ghc-default:
     environment:
       - STACK_FILE: "stack.yaml"
     <<: *defaults
 
   cabal:
     working_directory: ~/build
     docker:
-      - image: haskell:8.10.1
+      - image: haskell:8.10.2
```
Comment (Ailrun) [inline .circleci/config.yml:153]: Why do you use `8.10.2` instead of `8.10.3`?

**Review by Ailrun — APPROVED** [review]
LGTM.
BTW, I guess we should wait until Docker/github actions (or GHCup/hvr ppr) support 8.10.3, correct?
Otherwise, our CI will always give X for PRs/commits...

**Comment by jneira** [comment]
well, I think we could merge without the cabal circleci job, but a comment about we should update to 8.10.3 would be good

**Comment by Ailrun** [comment]
Make sense, could you add such a comment?

**Comment by jneira** [comment]
This test is failing for ubuntu and ghc-8.10.2:

```
 FAIL (6.06s)
      test/functional/Eval.hs:185:
      expected: "{- |\nTests in plain comments in both single line or multi line format, both forward and backward.\nTests are ignored if:\n    * do not start on the first column\n    * are in multi line comments that open and close on the same line\n-}\nmodule THaddock () where\n\n{- ORMOLU_DISABLE -}\n\n-- | Single line comment\n-- >>> \"a\"++\"b\"\n-- \"ab\"\n\n{- | Multi line comment\n\n>>> \"b\"++\"c\"\n\"bc\"\n-}\n\ndouble a = a + a\n-- ^ Single line backward comments\n-- >>> double 11\n-- 22\n\ntwice :: [a] -> [a]\ntwice a = a ++ a\n{- ^ Multi-line backward comments\n>>> twice \"ABC\"\n\"ABCABC\"\n-}\n\n{- | >>> 2+five\n7\n\n   ^-- This works, as it starts at the first column after the header.\n\n     >>> IGNORED as it does not start on the first column\n-}\nfive = 5\n"
       but got: "{- |\nTests in plain comments in both single line or multi line format, both forward and backward.\nTests are ignored if:\n    * do not start on the first column\n    * are in multi line comments that open and close on the same line\n-}\nmodule THaddock () where\n\n{- ORMOLU_DISABLE -}\n\n-- | Single line comment\n-- >>> \"a\"++\"b\"\n-- \"ab\"\n\n{- | Multi line comment\n\n>>> \"b\"++\"c\"\n\"bc\"\n-}\n\ndouble a = a + a\n-- ^ Single line backward comments\n-- >>> double 11\n-- 22\n\ntwice :: [a] -> [a]\ntwice a = a ++ a\n{- ^ Multi-line backward comments\n>>> twice \"ABC\"\n<BLANKLINE>\nByteCodeLink.lookupCE\nDuring interactive linking, GHCi couldn't find the following symbol:\n  interactive_Ghci1_asPrint_closure\nThis may be due to you not asking GHCi to load extra object files,\narchives or DLLs needed by your current session.  Restart GHCi, specifying\nthe missing library using the -L/path/to/object/dir and -lmissinglibname\nflags, or simply by naming the relevant files on the GHCi command line.\nAlternatively, this link failure might indicate a bug in GHCi.\nIf you suspect the latter, please report this as a GHC bug:\n  https://www.haskell.org/ghc/reportabug\n-}\n\n{- | >>> 2+five\n7\n\n   ^-- This works, as it starts at the first column after the header.\n\n     >>> IGNORED as it does not start on the first column\n-}\nfive = 5\n"
```

**Comment by jneira** [comment]
The error seems to be transient, it was present in https://github.com/haskell/haskell-language-server/runs/1613921454

**Comment by Anrock** [comment]
@jneira sorry, what are we blocked with exactly? I just updated my ghc and realized hls is not yet working with it.

**Comment by jneira** [comment]
@Anrock we are waiting for the update of github actions envs which includes ghc-8.10.3 (see the issue [here](https://github.com/actions/virtual-environments/issues/2353)) as commented in https://github.com/haskell/haskell-language-server/issues/311 (i did not link the issue, sorry)

**Comment by jneira** [comment]
closing, as ghc-8.10.3 support has been added by #721

**Pull request closed without merging**